src/EnergyPlus/ChillerElectricASHRAE205.cc

```diff
@@ -0,0 +1,1665 @@
+#include <EnergyPlus/FaultsManager.hh>
+#include <EnergyPlus/FluidProperties.hh>
+#include <EnergyPlus/General.hh>
+#include <EnergyPlus/GeneralRoutines.hh>
+#include <EnergyPlus/GlobalNames.hh>
+#include <EnergyPlus/HeatBalanceInternalHeatGains.hh>
+#include <EnergyPlus/InputProcessing/InputProcessor.hh>
+#include <EnergyPlus/NodeInputManager.hh>
+#include <EnergyPlus/OutAirNodeManager.hh>
+#include <EnergyPlus/OutputProcessor.hh>
+#include <EnergyPlus/OutputReportPredefined.hh>
+#include <EnergyPlus/Plant/DataPlant.hh>
+#include <EnergyPlus/Plant/PlantLocation.hh>
+#include <EnergyPlus/PlantUtilities.hh>
+#include <EnergyPlus/ScheduleManager.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+
+namespace EnergyPlus::ChillerElectricASHRAE205 {
+
+constexpr std::array<std::string_view, static_cast<int>(AmbientTempIndicator::Num) - 1> AmbientTempNamesUC{
+    "SCHEDULE",
+    "ZONE",
+    "OUTDOORS",
+};
+
+constexpr std::array<std::string_view, static_cast<int>(PerformanceInterpolationType::Num)> InterpolationMethods{
+    "LINEAR",
+    "CUBIC",
+};
+
+void tk205ErrCallback(tk205::MsgSeverity message_type, const std::string &message, void *context_ptr)
+{
+    std::pair<EnergyPlusData *, std::string> contextPair = *(std::pair<EnergyPlusData *, std::string> *)context_ptr;
+    std::string fullMessage = contextPair.second + ": " + message;
+    if (message_type == tk205::MsgSeverity::ERR_205) {
+        ShowSevereError(*contextPair.first, fullMessage);
+        ShowFatalError(*contextPair.first, "libtk205: Errors discovered, program terminates.");
+    } else {
+        if (message_type == tk205::MsgSeverity::WARN_205) {
+            ShowWarningError(*contextPair.first, fullMessage);
+        } else if (message_type == tk205::MsgSeverity::INFO_205) {
+            ShowMessage(*contextPair.first, fullMessage);
+        } else {
+            ShowMessage(*contextPair.first, fullMessage);
+        }
+    }
+}
+
+void getChillerASHRAE205Input(EnergyPlusData &state)
+{
+    static constexpr std::string_view RoutineName("getChillerASHRAE205Input: "); // include trailing blank space
+
+    using namespace tk205;
+    RSInstanceFactory::register_factory("RS0001", std::make_shared<RS0001Factory>());
+
+    bool ErrorsFound{false};
+
+    state.dataIPShortCut->cCurrentModuleObject = ChillerElectricASHRAE205::ASHRAE205ChillerSpecs::ObjectType;
+    auto &ip = state.dataInputProcessing->inputProcessor;
+    int numElectric205Chillers = ip->getNumObjectsFound(state, state.dataIPShortCut->cCurrentModuleObject);
+
+    if (numElectric205Chillers <= 0) {
+        ShowSevereError(state, format("No {} equipment specified in input file", state.dataIPShortCut->cCurrentModuleObject));
+        ErrorsFound = true;
+    }
+
+    state.dataChillerElectricASHRAE205->Electric205Chiller.allocate(numElectric205Chillers);
+
+    auto const ChillerInstances = ip->epJSON.find(state.dataIPShortCut->cCurrentModuleObject).value();
+    auto ChillerNum{0};
```

Comment:
This is being created as an `int`, so the `auto` shorthand is really not doing anything valuable here.  Probably not worth changing if there's no other changes.

Reply:
Fixed.